packages/dcos-integration-test/extra/test_endpoints.py

```diff
@@ -116,15 +117,24 @@ def test_if_pkgpanda_metadata_is_available(cluster):
 
 
 def test_if_dcos_history_service_is_getting_data(cluster):
-    r = cluster.get('/dcos-history-service/history/last')
-    assert r.status_code == 200
-    # Make sure some basic fields are present from state-summary which the DC/OS
-    # UI relies upon. Their exact content could vary so don't test the value.
-    json = r.json()
-    assert 'cluster' in json
-    assert 'frameworks' in json
-    assert 'slaves' in json
-    assert 'hostname' in json
+    @retry(stop_max_delay=20000, wait_fixed=500)
+    def check_up():
+        r = cluster.get('/dcos-history-service/history/last')
+        assert r.status_code == 200
+        # Make sure some basic fields are present from state-summary which the DC/OS
+        # UI relies upon. Their exact content could vary so don't test the value.
+        json = r.json()
```

Comment:
`assert {'cluster', 'frameworks', 'slaves', 'hostname'} <= json.keys()`

That way we get a nice structured error message with a diff of the dictionaries if there's an issue.